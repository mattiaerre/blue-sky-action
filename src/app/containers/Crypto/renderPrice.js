import numeral from 'numeral';
import PropTypes from 'prop-types';
import React from 'react';

function renderPrice({ amount, color, foregroundColor, name }) {
  const formatted = numeral(amount).format('0,0.00');
  return (
    <div
      className="Price card"
      style={{
        backgroundColor: color,
        color: foregroundColor
      }}
    >
      <div className="card-body">
        <span>
          <span>$</span>
          <span>{formatted.split('.')[0]}</span>
          <span>.{formatted.split('.')[1]}</span>
        </span>
        <small>{name.toUpperCase()} PRICE</small>
      </div>
    </div>
  );
}

renderPrice.propTypes = {
  amount: PropTypes.number.isRequired,
  color: PropTypes.string.isRequired,
  foregroundColor: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

export default renderPrice;
