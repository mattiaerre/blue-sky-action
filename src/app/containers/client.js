import { Lokka } from 'lokka';
import { Transport } from 'lokka-transport-http';

const client = baseUrl =>
  new Lokka({
    transport: new Transport(`${baseUrl}/graphql`)
  });

export default client;
