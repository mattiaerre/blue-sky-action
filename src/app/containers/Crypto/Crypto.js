import React, { Component } from 'react';
import numeral from 'numeral';
import client from '../client';

const query = `
query SpotPrices {
  spotPrices(currency: "USD") {
    amount
    code
    color
    currency
    foregroundColor
    name
  }
}
`;

function renderPrice({ amount, color, foregroundColor, name }) {
  const formatted = numeral(amount).format('0,0.00');
  return (
    <div
      className="Price card"
      style={{
        backgroundColor: color,
        color: foregroundColor
      }}
    >
      <div className="card-body">
        <span>
          <span>$</span>
          <span>{formatted.split('.')[0]}</span>
          <span>.{formatted.split('.')[1]}</span>
        </span>
        <small>{name.toUpperCase()} PRICE</small>
      </div>
    </div>
  );
}

class Crypto extends Component {
  constructor(props) {
    super(props);
    this.fetchSpotPrices();
    this.timer = setInterval(this.fetchSpotPrices, 15 * 1000);
  }

  state = {
    loading: true,
    spotPrices: []
  };

  componentWillUnmount() {
    clearInterval(this.timer);
  }

  fetchSpotPrices = () => {
    client.query(query).then(data => {
      this.setState({ loading: false, spotPrices: data.spotPrices });
    });
  };

  render() {
    const { loading, spotPrices } = this.state;
    return (
      <div className={`row ${loading ? 'loading' : ''}`}>
        {spotPrices.map(spotPrice => (
          <div
            className="col-lg-3 col-md-6 col-sm-12 mb-4"
            key={spotPrice.code}
          >
            {renderPrice(spotPrice)}
          </div>
        ))}
      </div>
    );
  }
}

export default Crypto;
