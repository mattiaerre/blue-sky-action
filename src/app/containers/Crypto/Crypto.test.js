import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Crypto from './Crypto';
import client from '../client';

configure({ adapter: new Adapter() });

jest.mock('../client');

const spotPrices = [
  {
    amount: 16503.51,
    code: 'BTC',
    color: '#FFB119',
    currency: 'USD',
    name: 'Bitcoin'
  },
  {
    amount: 2609.98,
    code: 'BCH',
    color: '#8DC451',
    currency: 'USD',
    name: 'Bitcoin Cash'
  },
  {
    amount: 1078.23,
    code: 'ETH',
    color: '#6F7CBA',
    currency: 'USD',
    name: 'Ethereum'
  },
  {
    amount: 278.8,
    code: 'LTC',
    color: '#B5B5B5',
    currency: 'USD',
    name: 'Litecoin'
  }
];

describe('<Crypto />', () => {
  test('to match snapshot', () => {
    const promise = Promise.resolve({
      spotPrices
    });
    client.query = jest.fn(() => promise);

    const wrapper = mount(<Crypto />);

    return promise
      .then(() => {
        expect(wrapper.debug()).toMatchSnapshot();
        wrapper.update();
      })
      .then(() => {
        expect(wrapper.debug()).toMatchSnapshot();
        wrapper.unmount();
      });
  });
});
