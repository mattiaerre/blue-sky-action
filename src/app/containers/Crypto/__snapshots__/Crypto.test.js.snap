// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Crypto /> to match snapshot 1`] = `
"<Crypto>
  <div className=\\"row loading\\" />
</Crypto>"
`;

exports[`<Crypto /> to match snapshot 2`] = `
"<Crypto>
  <div className=\\"row \\">
    <div className=\\"col-lg-3 col-md-6 col-sm-12 mb-4\\">
      <div className=\\"Price card\\" style={{...}}>
        <div className=\\"card-body\\">
          <span>
            <span>
              $
            </span>
            <span>
              16,503
            </span>
            <span>
              .
              51
            </span>
          </span>
          <small>
            BITCOIN
             PRICE
          </small>
        </div>
      </div>
    </div>
    <div className=\\"col-lg-3 col-md-6 col-sm-12 mb-4\\">
      <div className=\\"Price card\\" style={{...}}>
        <div className=\\"card-body\\">
          <span>
            <span>
              $
            </span>
            <span>
              2,609
            </span>
            <span>
              .
              98
            </span>
          </span>
          <small>
            BITCOIN CASH
             PRICE
          </small>
        </div>
      </div>
    </div>
    <div className=\\"col-lg-3 col-md-6 col-sm-12 mb-4\\">
      <div className=\\"Price card\\" style={{...}}>
        <div className=\\"card-body\\">
          <span>
            <span>
              $
            </span>
            <span>
              1,078
            </span>
            <span>
              .
              23
            </span>
          </span>
          <small>
            ETHEREUM
             PRICE
          </small>
        </div>
      </div>
    </div>
    <div className=\\"col-lg-3 col-md-6 col-sm-12 mb-4\\">
      <div className=\\"Price card\\" style={{...}}>
        <div className=\\"card-body\\">
          <span>
            <span>
              $
            </span>
            <span>
              278
            </span>
            <span>
              .
              80
            </span>
          </span>
          <small>
            LITECOIN
             PRICE
          </small>
        </div>
      </div>
    </div>
  </div>
</Crypto>"
`;
